podaac/subsetter/subset.py

```diff
@@ -709,34 +709,71 @@ def subset_with_bbox(dataset, lat_var_names, lon_var_names, time_var_names, bbox
         oper = operator.or_
 
     datasets = []
-    for lat_var_name, lon_var_name, time_var_name in zip(
-            lat_var_names, lon_var_names, time_var_names
-    ):
-        if GROUP_DELIM in lat_var_name:
-            var_prefix = GROUP_DELIM.join(lat_var_name.strip(GROUP_DELIM).split(GROUP_DELIM)[:-1])
-            group_vars = [
-                var for var in dataset.data_vars.keys()
-                if var.startswith(f'{GROUP_DELIM}{var_prefix}')
-            ]
-        else:
-            group_vars = list(dataset.keys())
-
-        group_dataset = dataset[group_vars]
-
-        # Calculate temporal conditions
-        temporal_cond = build_temporal_cond(min_time, max_time, group_dataset, time_var_name)
-        group_dataset = xre.where(
-            group_dataset,
-            oper(
-                (group_dataset[lon_var_name] >= lon_bounds[0]),
-                (group_dataset[lon_var_name] <= lon_bounds[1])
-            ) &
-            (group_dataset[lat_var_name] >= lat_bounds[0]) &
-            (group_dataset[lat_var_name] <= lat_bounds[1]) &
-            temporal_cond,
-            cut
-        )
-        datasets.append(group_dataset)
+    if variables:
```

Comment:
You're repeating a lot of code in the if/else branches, and there's only one small change between the two branches. Can you instead refactor this so the code isn't repeated and the only code in the if/else is whatever is specifically needed when variables are provided?

Reply:
Part of the difficulty is the loop is based off of the lat/lon names. So if we only want one variable, but two lat/lon variables are present, it goes through the loop a second time and an error occurs later in recombine. Though a small change is needed the loops are different so that one of the loops is zipped to the variable names if variables are imported at all. Another restructure would be to only have the variable extension happen once and not do it again even if there are other lat/lons added.

Reply:
@skorper I think this restructure makes the most sense. We need to add the extra variable for cases like OCO3 but can't over add when the variable needs to match it's respective lat/lon.